osu.Game/Rulesets/Scoring/HitResult.cs

```diff
@@ -350,6 +350,9 @@ public static void ValidateHitResultPair(HitResult maxResult, HitResult minResul
             if (maxResult.IsBonus() && minResult != HitResult.IgnoreMiss)
                 throw new ArgumentOutOfRangeException(nameof(minResult), $"{HitResult.IgnoreMiss} is the only valid minimum result for a {maxResult} judgement.");
 
+            if (minResult == HitResult.IgnoreMiss)
+                return;
```

Comment:
Should probably update `HitResultTest.TestValidResultPairs()` with coverage of this.